it/uk/gov/hmrc/minorentityidentificationfrontend/controllers/ControllersSupportWelshTranslationISpec.scala

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.minorentityidentificationfrontend.controllers
+
+import org.jsoup.nodes.Document
+import org.scalatest.prop.TableDrivenPropertyChecks.forAll
+import org.scalatest.prop.{TableFor1, TableFor2, Tables}
+import play.api.test.Helpers.{OK, await, defaultAwaitTimeout}
+import uk.gov.hmrc.minorentityidentificationfrontend.assets.TestConstants._
+import uk.gov.hmrc.minorentityidentificationfrontend.featureswitch.core.config.{EnableFullTrustJourney, FeatureSwitching}
+import uk.gov.hmrc.minorentityidentificationfrontend.stubs.{AuthStub, StorageStub}
+import uk.gov.hmrc.minorentityidentificationfrontend.utils.ComponentSpecHelper
+import uk.gov.hmrc.minorentityidentificationfrontend.utils.ViewSpecHelper.ElementExtensions
+
+class ControllersSupportWelshTranslationISpec
+  extends ComponentSpecHelper
+    with StorageStub
+    with AuthStub
+    with FeatureSwitching{
+
+  private def baseUrl = "/identify-your-trust"
+
+  "all the views requested via GET" should {
+
+    object TestPrecondition {
+      type TestPrecondition = () => Any
+      val checkYourAnswers: TestPrecondition = () => stubRetrieveEntityDetails(testJourneyId)(OK, testTrustJourneyDataJson)
+      val confirmSautr: TestPrecondition = () => stubRetrieveUtr(testJourneyId)(status = OK, body = testSautrJson)
+      val confirmChrn: TestPrecondition = () => stubRetrieveCHRN(testJourneyId)(status = OK, testCHRN)
+      val confirmSaPostcode: TestPrecondition = () => stubRetrievePostcode(testJourneyId)(status = OK, testSaPostcode)
+    }
+
+    val allGETUrlsToBeTested: TableFor2[String, TestPrecondition.TestPrecondition] =
```

Comment:
You could also test the CannotConfirmBusinessController.

Reply:
done